packages/nuxt/src/components/plugins/loader.ts

```diff
@@ -46,10 +47,14 @@ export const LoaderPlugin = (options: LoaderOptions) => createUnplugin(() => {
       s.replace(/(?<=[ (])_?resolveComponent\(\s*["'](lazy-|Lazy(?=[A-Z]))?([^'"]*)["'][^)]*\)/g, (full: string, lazy: string, name: string) => {
         const component = findComponent(components, name, options.mode)
         if (component) {
-          // @ts-expect-error TODO: refactor to nuxi
-          if (component._internal_install && tryUseNuxt()?.options.test === false) {
-            // @ts-expect-error TODO: refactor to nuxi
-            import('../../core/features').then(({ installNuxtModule }) => installNuxtModule(component._internal_install))
+          // TODO: refactor to nuxi
+          const internalInstall = ((component as any)._internal_install) as string
+          if (internalInstall && nuxt?.options.test === false) {
+            if (!nuxt.options.dev) {
+              const relativePath = relative(nuxt.options.rootDir, id)
+              throw new Error(`[nuxt] \`~/${relativePath}\` is using \`${component.pascalName}\` which requires \`${internalInstall}\``)
```

Comment:
What if the component is registred at runtime (in a plugin). Would we still strictly fail builds?

Reply:
Yes. Because the transform has already taken place and injected `<NuxtImg>` pointing to the stub component provided by Nuxt, and if used it would throw an error:

https://github.com/nuxt/nuxt/blob/f209158352b09d1986aa320e29ff36353b91c358/packages/nuxt/src/app/components/nuxt-stubs.ts#L3-L17

Reply:
Ah i see it is really there!